crates/sui-framework/sources/hex.move

```diff
@@ -23,6 +30,27 @@ module sui::hex {
         r
     }
 
+    /// Decode hex into `bytes`
```

Comment:
Can you add some detail to these docs, i.e.

- The fact that this is case insensitive, and does not expect a leading `0x`.
- The cases in which the function aborts (invalid hex character, and odd length input).